### ISSUE TITLE
JOSS review issues

### ISSUE THREAD

**Issue by dgrtwo** [andyyao95/walkr#2]
Regarding the [review for JOSS](https://github.com/openjournals/joss-reviews/issues/61). Many thanks for your patience, I know it has taken me a long time to reach this review!

Overall this is a great package and I've been impressed as I've played with the mathematical sampling. I particularly like the detail and figures in the vignette and am impressed with the test coverage, and I'll have checked off all but two boxes in my review.

Overall there is one major issue and one minor; once we've addressed them I'd be happy to recommend acceptance.

## General checks

As you're probably aware the current commit on GitHub doesn't currently pass R CMD CHECK or its unit tests; and the release currently on CRAN (0.3.4) doesn't have an equivalent in the code. Examples of issues include:

  * Since some functionality has changed in that time (e.g. adding an argument to dikin_walk), unit tests no longer pass
  * Packages cem, MatchIt, and Matching are imported but are not in IMPORTS, which prevents running after installation

This could lead to open source challenges, since people wanting to contribute to the software can't install it from GitHub without additional package installation, and can't use unit tests to check they suggest changes (furthermore, it looks like the source code in CRAN has "diverged" with the latest 0.3.4 release, so it may not be clear how to fix existing issues). I wouldn't consider CRAN/GitHub changes a blocker if they didn't affect installation.

How soon were you planning on fixing the issues in the current GitHub version and submitting it to CRAN? Even if you don't feel the new vignettes are ready for CRAN we could get it to pass R CMD CHECK (I'm happy to help). If it's not on the near horizon we could ask @arfon for an opinion on whether it should still be accepted.

## Software paper

* By JOSS policy, the references in paper.bib need to list their DOIs when available. For example, your third link paper the DOI: 10.1287/moor.1110.0519

**Comment by andyyao95** [comment]
Hi @dgrtwo, 

Thank you for your time and effort in reviewing our paper! With regard to your comments: 

- I just updated the github version and it is now the same version as the one on CRAN (I should've done this earlier). The github version now should be able to pass all tests and CMD checks, as it did on my machine. Let me know if there is any issue with this. 

- I've updated the .bib file, including all the DOIs I could find, except one which is a reference to an R package. 

Thanks again. Let me know if you have any other questions. Looking forward to hearing back from you. 

Best,
Andy
